### PR TITLE
feat(chip): add small size

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

Chip does not have the small size. There also wasn't a size prop to show that large is the default for the ionic theme.

## What is the new behavior?

- Added the small size
- Added the size prop, large being the default for ionic
- Added tests

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Preview](https://ionic-framework-git-rou-5476-ionic1.vercel.app/src/components/chip/test/size?ionic:theme=ionic)

### CODE REVIEW

File: core/src/components/chip/chip.ionic.scss [hunk 1]
```diff
@@ -9,14 +10,13 @@ $ionic-states-hover: #{rgba(#05080f, 0.16)}; // We should review how to make thi
...
   --background: #{globals.$ionic-color-neutral-10};
-  --border-color: transparent;
   --border-radius: #{globals.$ionic-border-radius-rounded-small};
   --color: #{globals.$ionic-color-neutral-900};
   --focus-ring-color: #{$ionic-states-focus-primary};
   --focus-ring-width: #{globals.$ionic-border-size-medium};
 
   @include globals.font-smoothing;
-  @include globals.padding(globals.$ionic-space-xs, globals.$ionic-space-xxs);
+  @include globals.padding(globals.$ionic-space-xxs, globals.$ionic-space-xs);
```
Comment (thetaPC) [inline core/src/components/chip/chip.ionic.scss:19]: Verified on Figma that these values had to be swapped.

File: core/src/components/chip/test/shape/chip.e2e.ts-snapshots/chip-default-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (thetaPC) [inline core/src/components/chip/test/shape/chip.e2e.ts-snapshots/chip-default-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: The non-size screenshots that were updated is due to the padding being updated.

**Review by brandyscarney — CHANGES_REQUESTED** [review]
Can we update the PR title / commit message to include that we are now setting the default shape to `"large"` for the ionic theme?

File: core/src/components/chip/chip.ionic.scss (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/chip/chip.ionic.scss:18]: Double verified 👍

File: core/src/components/chip/chip.ionic.scss [hunk 2]
```diff
@@ -105,3 +99,18 @@ $ionic-states-hover: #{rgba(#05080f, 0.16)}; // We should review how to make thi
...
+  min-height: 24px;
+
+  font-size: #{tokens.$ionic-font-size-s};
+}
+
+:host(.chip-large) {
+  min-height: 32px;
+
+  font-size: globals.$ionic-font-size-m;
+}
```
[the remark this answers is not in the document]
Comment (thetaPC) [reply core/src/components/chip/chip.ionic.scss:116]: The globals was just a copy and paste from an earlier implementation. I plan on opening another PR that cleans up the file to follow the structure we have for ios and md. Let me know if you want me to just one one import in this PR.

[the remark this answers is not in the document]
Comment (brandyscarney) [reply core/src/components/chip/chip.ionic.scss:116]: Oh I see! A new PR to clean it up works for me. 👍

**Review by brandyscarney — APPROVED** [review]
LGTM!

**Pull request merged**